spec.yaml

```diff
@@ -262,55 +274,84 @@ definitions:
       senderDisplayName:
         type: string
         description: |
-          Display name of the owner of the resource
+          Display name of the user that wants to share the resource
         example: John Doe
       shareType:
         type: string
+        enum: ["user", "group"]
         description: |
-          Share type (user or group share)
+          Recipient share type
         example: user
       resourceType:
         type: string
         description: |
-          Resource type (file, calendar, contact, ...)
-        example: file
-      protocol:
```

Comment:
This looks like a breaking change?
Servers adhering to this spec will not be able to send shares to servers adhering to the previous spec, and vice versa.

Reply:
Just like renaming 'providerId' to 'resourceId', renaming 'protocol' to 'protocols' is a complicated project.
We probably need to at least make protocol versions discoverable?

Reply:
For the resource type, if `file` "includes" `folder` (which I guess is the case for the existing implementations) then the change is just a clarification. We may well leave it as it was before, that is implicitly saying that folders have type = `file`.

For `protocol` -> `protocols`, this is THE breaking change - and here we already broke compatibility with cs3org/OCM-API#54. I've updated the description of the PR to explicitly state that. What do you mean by making "protocol versions" discoverable?

Reply:
I mean since https://github.com/cs3org/OCM-API/pull/37 we have `apiVersion "1.0-proposal1" in https://cloud.pondersource.com/ocm-provider/ (although I think this is not documented in the spec itself), we should probably tag OCM 2.0 and change that `apiVersion` string to "2.0".
We can create PRs on OC-10 and on NC to make the necessary changes, while keeping backwards compatibility with OCM 1.0.

Reply:
Reva doesn't really need to stay compatible with OCM 1.0 because it doesn't support the share-with flow and save-to-your-cloud flow anyway.